src/System.Memory/src/System/Buffers/Text/Utf8Parser/Utf8Parser.Integer.Signed.N.cs

```diff
@@ -8,12 +8,186 @@ public static partial class Utf8Parser
     {
         private static bool TryParseSByteN(ReadOnlySpan<byte> text, out sbyte value, out int bytesConsumed)
         {
-            throw NotImplemented.ActiveIssue("https://github.com/dotnet/corefx/issues/24986");
+            if (text.Length < 1)
+                goto FalseExit;
+
+            int sign = 1;
+            int index = 0;
+            int c = text[index];
+            if (c == '-')
+            {
+                sign = -1;
+                index++;
+                if ((uint)index >= (uint)text.Length)
+                    goto FalseExit;
+                c = text[index];
+            }
+            else if (c == '+')
+            {
+                index++;
+                if ((uint)index >= (uint)text.Length)
+                    goto FalseExit;
+                c = text[index];
+            }
+
+            int answer;
+
+            // Handle the first digit (or period) as a special case. This ensures some compatible edge-case behavior with the classic parse routines
+            // (at least one digit must precede any commas, and a string without any digits prior to the decimal point must have at least 
+            // one digit after the decimal point.)
+            if (c == Utf8Constants.Period)
+                goto FractionalPartWithoutLeadingDigits;
+            if (!ParserHelpers.IsDigit(c))
+                goto FalseExit;
+            answer = c - '0';
+
+            for (; ; )
```

Comment:
nit: spacing, or just make this a while(true)

Reply:
also it seems all the parsing methods are identical except for the overflow checks.  Is there a way you can eliminate all the duplicate code?

Reply:
The 8-bit and 16-bit parsers could call the 32-bit parser and overlay their own range check. It trades off some performance though which isn't a tradeoff we generally make in this particular api area. We generally go for performance and accept that the code looks like it was generated by a bot.

Reply:
The spacing looks a bit weird, but it's the one that won't have everyone fighting VS's auto-formatting. And "for(;;)" seems to be the prevailing idiom over "while(true)" in this assembly.